packages/expo-image-picker/ios/MediaHandler.swift

```diff
@@ -368,14 +371,14 @@ internal struct MediaHandler {
   }
 
   private func handleVideo(from selectedVideo: PHPickerResult) async throws -> AssetInfo {
-    // Fast-path: If transcoding is disabled (passthrough) and we have direct access to the underlying
+    // Fast-path: Always try to use PHAssetResourceManager first to force iCloud downloads when we have direct access to the underlying
     // `PHAsset`, try to copy the original/full-size video resource via `PHAssetResourceManager`.
     // This avoids `loadFileRepresentation`, which can be noticeably slower once a user
     // tweaks only the metadata (e.g. adjusts the capture date) because the photo service marks the
     // asset as *adjusted* and will re-render a temporary file for us. Copying the resource bytes
     // ourselves is dramatically faster because it just streams the already-existing file.
 
-    if options.videoExportPreset == .passthrough, let assetId = selectedVideo.assetIdentifier {
```

Comment:
Can we still have the fast path when `passthrough` is enabled and we have access to the asset by detecting if the asset is in the cloud, and only then enable reading it from iCloud? Right now, we're making the API slower for almost everyone.